util/types/field_type.go

```diff
@@ -67,8 +67,8 @@ func NewFieldType(tp byte) *FieldType {
 // Aggregation is performed by MergeFieldType function.
 func AggFieldType(tps []*FieldType) *FieldType {
 	var currType FieldType
-	for _, t := range tps {
-		if currType.Tp == mysql.TypeUnspecified {
+	for i, t := range tps {
+		if i == 0 && currType.Tp == mysql.TypeUnspecified {
```

Comment:
yeah, the cases in the test file which I modified were wrong,
they cover this check.